tests/ha/test_multi_server_pod_failure.py

```diff
@@ -0,0 +1,287 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""
+HA test suite for Multiple (K) Server Pods Failure
+"""
+
+import logging
+import os
+import random
+import time
+from time import perf_counter_ns
+import pytest
+
+from config import CMN_CFG
+from config.s3 import S3_CFG
+from commons import constants as const
+from commons.ct_fail_on import CTFailOn
+from commons.errorcodes import error_handler
+from commons.helpers.health_helper import Health
+from commons.helpers.pods_helper import LogicalNode
+from commons.params import TEST_DATA_FOLDER
+from commons.utils import assert_utils
+from commons.utils import system_utils as sysutils
+from libs.di.di_mgmt_ops import ManagementOPs
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.prov.prov_k8s_cortx_deploy import ProvDeployK8sCortxLib
+from libs.s3.s3_multipart_test_lib import S3MultipartTestLib
+from libs.s3.s3_rest_cli_interface_lib import S3AccountOperations
+
+# Global Constants
+LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=R0902
+# 4pylint: disable=R090
+class TestMultiServerPodFailure:
+    """
+    Test suite for Multiple Server Pods Failure
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """
+        Setup operations for the test file.
+        """
+        LOGGER.info("STARTED: Setup Module operations.")
+        cls.csm_user = CMN_CFG["csm"]["csm_admin_user"]["username"]
+        cls.csm_passwd = CMN_CFG["csm"]["csm_admin_user"]["password"]
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.username = []
+        cls.password = []
+        cls.host_master_list = []
+        cls.node_master_list = []
+        cls.hlth_master_list = []
+        cls.host_worker_list = []
+        cls.node_worker_list = []
+        cls.pod_name_list = []
+        cls.ha_obj = HAK8s()
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.s3_clean = cls.test_prefix = cls.random_time = None
+        cls.s3acc_name = cls.s3acc_email = cls.bucket_name = cls.object_name = None
+        cls.restore_pod = cls.deployment_backup = cls.deployment_name = cls.restore_method = None
+        cls.deploy = cls.kvalue = None
+        cls.pod_dict = {}
+        cls.mgnt_ops = ManagementOPs()
```

Comment:
We don't need this right ? Are we going to test RC node shutdown test  for server pod too ?

Reply:
This is use for normal read/write too, hence we require this.